python/lsst/ts/maintel/standardscripts/set_dof.py

```diff
@@ -42,11 +55,140 @@ class SetDOF(ApplyDOF):
 
     """
 
+    def __init__(self, index):
+        super().__init__(index=index)
+
+    @classmethod
+    def get_schema(cls):
+        schema_yaml = """
+            $schema: http://json-schema.org/draft-07/schema#
+            $id: https://github.com/lsst-ts/ts_standardscripts/maintel/SetDOF.yaml
+            title: SetDOF v1
+            description: Configuration for SetDOF.
+            type: object
+            properties:
+              day:
+                description: Day obs to be used for synchronizing the state.
+                type: number
+                default: null
+              seq:
+                description: Sequence number to be used for synchronizing the state.
+                type: number
+                default: null
+            anyOf:
+                - required:
+                    - day
+                    - seq
+                - required:
+                    - dofs
+            additionalProperties: false
+        """
+        schema_dict = yaml.safe_load(schema_yaml)
+
+        base_schema_dict = super().get_schema()
+
+        for prop in base_schema_dict["properties"]:
+            schema_dict["properties"][prop] = base_schema_dict["properties"][prop]
+
+        return schema_dict
+
+    async def configure(self, config) -> None:
+        """Configure script.
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+        super(ApplyDOF, self).configure(config)
+
+        self.day = getattr(config, "day", None)
+        self.seq = getattr(config, "seq", None)
+
+    async def get_image_time(self, client, day, seq):
+        """Get the image time from the given day and sequence number.
+        Parameters
+        ----------
+        day : `int`
+            Day obs to be used for synchronizing the state.
+        seq : `int`
+            Sequence number to be used for synchronizing the state.
+        Returns
+        -------
+        end_time : `astropy.time.Time`
+            End time of the image.
+        Raises
+        ------
+        RuntimeError
+            Error querying time for image {day}:{seq}.
+        """
+        try:
+            query = f"""
+                SELECT  "imageDate", "imageNumber"
+                FROM "efd"."autogen"."lsst.sal.CCCamera.logevent_endReadout"
```

Comment:
This is using CCCamera. You might want to make this configurable and use MTCamera as default.